modules/config/main.tf

```diff
@@ -0,0 +1,304 @@
+locals {
+  cloudtrail_name = "${var.name}-config-trail"
+
+  cloudtrail_prefix      = var.cloudtrail_prefix == "" ? "gitlab-runner-config" : var.cloudtrail_prefix
+  cloudtrail_bucket      = length(var.cloudtrail_bucket) > 0 ? data.aws_s3_bucket.cloudtrail[0].bucket : "${var.name}-cloudtrail"
+  cloudtrail_bucket_name = length(var.cloudtrail_bucket) > 0 ? data.aws_s3_bucket.cloudtrail[0].bucket : aws_s3_bucket.cloudtrail[0].bucket
+  cloudtrail_bucket_arn  = "arn:aws:s3:::${local.cloudtrail_bucket}"
+
+  config_key         = var.config_key == "" ? "config.toml" : var.config_key
+  config_bucket      = length(var.config_bucket) > 0 ? data.aws_s3_bucket.config[0].bucket : "${var.name}-config"
+  config_bucket_name = length(var.cloudtrail_bucket) > 0 ? data.aws_s3_bucket.config[0].bucket : aws_s3_bucket.config[0].bucket
```

Comment:
This looks wrong, probably a copy+paste error? Surely the condition should be `length(var.config_bucket) > 0`, _not_ `cloudtrail_bucket`.